logic.js

```diff
@@ -0,0 +1,71 @@
+const Tweeter = function () {
+    
+    const posts = [ // array of objects that oild the posts id , texr and comments that is array himself
+        {
+            text: "First post!",
+            id: "p1",
+            comments: [
+                { id: "c1", text: "First comment on first post!" },
+                { id: "c2", text: "Second comment on first post!!" },
+                { id: "c3", text: "Third comment on first post!!!" }
+            ]
+        },
+        {
+            text: "Aw man, I wanted to be first",
+            id: "p2",
+            comments: [
+                { id: "c4", text: "Don't wory second poster, you'll be first one day." },
+                { id: "c5", text: "Yeah, believe in yourself!" },
+                { id: "c6", text: "Haha second place what a joke." }
+            ]
+        }
+    ]
+    let postIdCounter = 2 //count the number of total posts
+    let commentIdCounter = 6 //count the number of total comments
+
+    const getPosts = function () {
+        console.log(posts);
+        return posts //return all the posts array. log it if you want to see
+    }
+
+    const addPost = function (text) { // the function that handle the add post
+        postIdCounter++ //post id counter incremnt by 1
+        posts.push({ text: text, id: "p" + postIdCounter, comments: [] }) //pushing the new data to a new id and initial a comment array
+    } //void method (dont return nothing)
+
+    const removePost = function (id) { // the function htat handle removing post
+        for (let i in posts) { //runs on the posts array
+            if (posts[i].id === id) { // check if the parameter = the posts in position i id key
+                posts.splice(i, 1) //if so , delete and send i as the start point and delet only 1
+                break;//stop the code when you do that 
+            }
+        }
+        postIdCounter-- // post id counter sub by 1
```

Comment:
The 'postIdCounter' is a counter for the id, not for the array length, you should not decrease the idCounter because you might repeat the same id

Reply:
ok, go it.
thank you yoni

> The 'postIdCounter' is a counter for the id, not for the array length, you should not decrease the idCounter because you might repeat the same id

ok, got it.
thank's yoni

Reply:
ok , got it.
thank's yoni